addons/mail/static/src/discuss/call/common/call.js

```diff
@@ -108,16 +131,57 @@ export class Call extends Component {
         return raisingHandCards.concat(sessionCards, invitationCards);
     }
 
+    /** @returns {CardData[]} */
     get visibleMainCards() {
-        if (this.props.thread.activeRtcSession) {
-            return [
-                {
-                    key: "session_" + this.props.thread.activeRtcSession.id,
-                    session: this.props.thread.activeRtcSession,
-                },
-            ];
+        const activeSession = this.props.thread.activeRtcSession;
+        if (!activeSession) {
+            this.state.insetCard = undefined;
+            return this.visibleCards;
         }
-        return this.visibleCards;
+        switch (activeSession.videoStreams.size) {
+            case 1:
+                if (activeSession.videoStreams.get("screen") === activeSession.mainVideoStream) {
+                    this.setInset(activeSession);
+                }
+                if (
+                    activeSession.videoStreams.get("camera") &&
+                    activeSession.videoStreams.get("camera") === activeSession.mainVideoStream
+                ) {
+                    this.state.insetCard = undefined;
+                }
+                if (!activeSession.mainVideoStream && activeSession.videoStreams.get("screen")) {
+                    this.setInset(activeSession, "screen");
+                }
+                break;
+            case 2: {
+                const videoType =
+                    activeSession.videoStreams.get("screen") === activeSession.mainVideoStream
+                        ? "camera"
+                        : "screen";
+                this.setInset(activeSession, videoType);
+                break;
+            }
+        }
```

Comment:
This code looks incorrect: when only "screen" is enabled, it shows inset of self-card, as if there was a camera stream when there's none.
The inset should only be visible when there are 2 videoStreams for the active session

Reply:
Specs say to show inset even when only screen sharing, so code is ok